rewriter/Concern.cc

```diff
@@ -0,0 +1,152 @@
+#include "rewriter/Concern.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Names.h"
+#include "core/core.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+// Generates the following Ruby code:
+// extend T::Helpers
+void generateSendForExtendHelpers(core::LocOffsets loc, vector<ast::TreePtr> &stats) {
+    auto sendForExtendHelpers = ast::MK::Send1(loc, ast::MK::Self(loc), core::Names::extend(),
+                                               ast::MK::Constant(loc, core::Symbols::T_Helpers()));
+    stats.emplace_back(std::move(sendForExtendHelpers));
+}
+
+// Generates the following Ruby code:
+// mixes_in_class_methods(ClassMethods)
+void generateSendForMixesInClassMethods(core::LocOffsets loc, vector<ast::TreePtr> &stats) {
+    auto nameClassMethods =
+        ast::MK::UnresolvedConstant(loc, ast::MK::EmptyTree(), core::Names::Constants::ClassMethods());
+    auto sendForMixes =
+        ast::MK::Send1(loc, ast::MK::Self(loc), core::Names::mixesInClassMethods(), std::move(nameClassMethods));
+    stats.emplace_back(std::move(sendForMixes));
+}
```

Comment:
I admire your desire to factor code into small pieces, but I actually find this to be harder to read, because these functions are only called once (it looks like maybe one of these helpers is called twice). It means that instead of reading the code top-to-bottom, seeing which ast nodes are getting created and where, I have to jump all over this file. Also we're mutating a vector in the process, and that's hidden behind a function call.

Do you mind either:

- inlining all of these short functions into the place where they're called or,
- keeping them but making them return the thing they create, and doing the `stats.emplace_back(...)` calls inside the main method?